rpifwcrypto/rpifwcrypto.c

```diff
@@ -26,24 +26,33 @@
 /* VideoCore mailbox error flag */
 #define VC_MAILBOX_ERROR 0x80000000
 
+/* Crypto-related mailbox tags */
+typedef enum {
+    TAG_GET_CRYPTO_LAST_ERROR      = 0x0003008e,    // Get last error code
+    TAG_GET_CRYPTO_NUM_OTP_KEYS    = 0x0003008f,    // Get number of available OTP keys
+    TAG_GET_CRYPTO_KEY_STATUS      = 0x00030090,    // Get key status
+    TAG_SET_CRYPTO_KEY_STATUS      = 0x00038090,    // Set key status
+    TAG_GET_CRYPTO_ECDSA_SIGN      = 0x00030091,    // Sign data using ECDSA
+    TAG_GET_CRYPTO_HMAC_SHA256     = 0x00030092,    // Compute HMAC-SHA256
+    TAG_GET_CRYPTO_PUBKEY          = 0x00030093,    // Get public key
+} RPI_FW_CRYPTO_TAG;
+
 /* Common header structure for firmware mailbox messages */
 struct firmware_msg_header {
     uint32_t buf_size;
     uint32_t code;
     uint32_t tag;
     uint32_t tag_buf_size;
     uint32_t tag_req_resp_size;
-};
+} __attribute__((packed));
```

Comment:
Does `packed` do anything with what is essentially `uint32_t hdr[5]`? I think it should be unnecessary.

Reply:
Yes, it's redundant here. I just applied it to all the message headers but this will always be a fixed uint32_t structure

Reply:
I didn't notice a single message header that would need it - the trick is to define the structures so the natural packing is correct.

Reply:
So why all the `packed`s?

Reply:
Harmless future proofing and enabling some possibly none word aligned data but the vc impl no longer required that. I've removed packed to avoid confusion until if/when it's required.